### PR TITLE
Add interrupt support for Nordic GPIO.

### PR DESCRIPTION
**Pull request by simonjwright**
* arch/ARM/Nordic/drivers/nrf_common/nrf-gpio-tasks_and_events.ad[sb]:
      Added subprograms to enable, test for presence, acknowledge and
      disable both channel and port interrupts.

### CODE REVIEW

File: arch/ARM/Nordic/drivers/nrf_common/nrf-gpio-tasks_and_events.adb [hunk 1]
```diff
@@ -34,6 +34,31 @@ with HAL;              use HAL;
...
+   procedure Acknowledge_Port_Interrupt is
+   begin
+      GPIOTE_Periph.EVENTS_PORT := 0;
+   end Acknowledge_Port_Interrupt;
+
+   --------------------------
+   -- Channel_Event_Is_Set --
+   --------------------------
+
+   function Channel_Event_Is_Set (Chan : GPIOTE_Channel) return Boolean is
```
Comment (Fabien-Chouteau) [inline arch/ARM/Nordic/drivers/nrf_common/nrf-gpio-tasks_and_events.adb:59]: Not a huge deal but we usually don't add a `is` in the names of test functions like that.
Also we put the `is` keyword on a new line:
```ada
function Channel_Event_Set (Chan : GPIOTE_Channel) return Boolean
is (GPIOTE_Periph.EVENTS_IN (Integer (Chan)) /= 0);
```

**Review by Fabien-Chouteau — COMMENTED** [review]
Thank you @simonjwright !

Just one comment.

File: arch/ARM/Nordic/drivers/nrf_common/nrf-gpio-tasks_and_events.adb (hunk 1 above)
Comment (simonjwright) [reply arch/ARM/Nordic/drivers/nrf_common/nrf-gpio-tasks_and_events.adb]: Added another commit to sort this.

After a bit more poking about, I see that I could have used `nRF.Events`. You might think that my effort could have been better spent adding how-to comments to that package?

By the way, the comments in nrf-events.adb mention the nRF series RM section 9; in nRF51_RM_v3.0, this is section 10.

**Review by Fabien-Chouteau — APPROVED** [review]

**Comment by Fabien-Chouteau** [comment]
Thank you @simonjwright

**Pull request merged**